routes/error_codes.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module ErrorCodes
```

Comment:
The name `ErrorCodes` appears too generic to me. The usage, `fail ErrorCodes::DependencyError`, doesn't feel intuitive. We could consider using a name like `CloverError`, similar to `RodaError` or `Sequel::Error`.